components/auth/src/main/kotlin/org/cloudfoundry/credhub/auth/UserContextFactory.kt

```diff
@@ -7,70 +7,32 @@ import org.springframework.security.oauth2.server.resource.authentication.JwtAut
 import org.springframework.security.web.authentication.preauth.PreAuthenticatedAuthenticationToken
 import org.springframework.stereotype.Component
 import java.security.cert.X509Certificate
+import java.time.Instant
 
 @Component
 class UserContextFactory {
     @Autowired(required = false)
-//    private val resourceServerTokenServices: ResourceServerTokenServices? = null
-
-    /*
-     * The "iat" and "exp" claims are parsed by Jackson as integers,
-     * because JWT defines these as seconds since Epoch
-     * (https://tools.ietf.org/html/rfc7519#section-2). That means it has a
-     * Year-2038 bug. To adapt to our local model, hoping JWT will some day be improved,
-     * this function returns a numeric value as long.
-     */
-    private fun claimValueAsLong(additionalInformation: Map<String, Any>): Long =
-        (additionalInformation["iat"] as Number).toLong()
 
     fun createUserContext(authentication: Authentication?): UserContext =
         if (authentication is PreAuthenticatedAuthenticationToken) {
             createUserContext(authentication)
         } else {
-            createUserContext(authentication as JwtAuthenticationToken, null)
+            createUserContext(authentication as JwtAuthenticationToken)
         }
 
-    fun createUserContext(
-        authentication: JwtAuthenticationToken,
-        maybeToken: String?,
-    ): UserContext {
+    fun createUserContext(authentication: JwtAuthenticationToken): UserContext {
         val jwt: Jwt = authentication.principal as Jwt
         val claims = jwt.claims
         val grantType = claims["grant_type"] as String
         val clientId = claims["client_id"] as String
         val userId = claims["user_id"] as String?
         val userName = claims["user_name"] as String?
         val issuer = claims["iss"] as String?
-        val validFrom: Long = 0
-        val validUntil: Long = 0
+        val validFrom: Long = (claims["iat"] as Instant).toEpochMilli()
+        val validUntil: Long = (claims["exp"] as Instant).toEpochMilli()
```

Comment:
I think that both of these need to be:
```
(claims["iat"] as Instant).epochSecond
```

Both "MUST be a number containing a NumericDate value" per the [RFC](https://tools.ietf.org/html/rfc7519#section-2) mentioned in the deleted comment (line number 16). And `NumericDate` definition in the RFC is as follows: 
> NumericDate
      A JSON numeric value representing the number of seconds from
      1970-01-01T00:00:00Z UTC until the specified UTC date/time,
      ignoring leap seconds.